### ISSUE TITLE
Can this FreeRTOS port also be used for CV32E40S?

### ISSUE THREAD

**Issue by ChristianLarmann** [openhwgroup/core-v-freertos#3]
This project provides FreeRTOS for the core-v-mcu, which incorporates the CV32E40P. I want to run FreeRTOS on the E40S now. Is this compatible and if not, what has to be adapted?

For example, I have seen for the RI5CY core six additional registers have to be stored for each task (lpstart0, lpend0, lpcount0...). Therefore, it would be good to know if something similar during the context switches must be considered for the E40S.

**Comment by DBees** [comment]
@n9wxu @promodkumar-ashling Could you please give guidance to @ChristianLarmann for how to adapt FreeRTOS for an E40 core that has additional registers?

**Comment by n9wxu** [comment]
The risc-v port is designed to support extensions. Take a look at the gcc/risc-v/chip_specific_extensions Joseph JulicherOn Feb 14, 2024, at 7:34 PM, DBees ***@***.***> wrote:﻿
@n9wxu @promodkumar-ashling Could you please give guidance to @ChristianLarmann for how to adapt FreeRTOS for an E40 core that has additional registers?

**Comment by ChristianLarmann** [comment]
Yes, there are multiple extensions available but to me it is not clear which one I should use as the CV32E40S has the CORE-V instruction set extension as the E40P but also the XSecure extension. Do I need to take the RV32I_CLINT_no_extensions one and adapt it if necessary?
